chain-tx-enclave/tx-query-next/enclave-app/src/handler/decryption_request.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    io::{Read, Write},
+    net::TcpStream,
+    sync::{Arc, Mutex},
+};
+
+use parity_scale_codec::{Decode, Encode};
+use secp256k1::{key::PublicKey, Secp256k1};
+use zeroize::Zeroize;
+
+use chain_core::{
+    common::H256,
+    state::account::WithdrawUnbondedTx,
+    tx::{
+        data::{access::TxAccessPolicy, attribute::TxAttributes, Tx},
+        TxWithOutputs,
+    },
+};
+use enclave_protocol::{DecryptionRequest, DecryptionResponse, EnclaveRequest, EnclaveResponse};
+use enclave_utils::SealedData;
+
+pub fn get_random_challenge() -> H256 {
+    rand::random()
+}
+
+pub fn verify_decryption_request(decryption_request: &DecryptionRequest, challenge: H256) -> bool {
+    !decryption_request
+        .verify(&Secp256k1::verification_only(), challenge)
+        .is_err()
+}
+
+pub fn handle_decryption_request(
+    decryption_request: &DecryptionRequest,
+    zmq_stream: Arc<Mutex<TcpStream>>,
+) -> Result<DecryptionResponse, String> {
+    let enclave_request = EnclaveRequest::GetSealedTxData {
+        txids: decryption_request.body.txs.clone(),
+    }
+    .encode();
+
+    // TODO: check tx size
+    let max_result_len = decryption_request.body.txs.len() * 8000;
+    let mut result_buf = vec![0u8; max_result_len + 1];
```

Comment:
is this necessary? this was a temporary ugly hack in the original due to the ocall result buffer fiddling... here one could just keep reading off the stream and allocate as necessary? it may need some small change in `ZmqStreamHelper`

Reply:
Ok. I'll replace it with [IoReader](https://docs.rs/parity-scale-codec/1.3.0/parity_scale_codec/struct.IoReader.html)

Reply:
Unfortunately, we cannot do this using `IoReader` because `rustls::Stream` does not implement `std::io::Seek` which is required by `IoReader`.

Reply:
> here one could just keep reading off the stream and allocate as necessary

We need to know the number of bytes we want to read from a stream. `IoReader` handled it internally but unfortunately, we cannot use that here.

Reply:
I meant in the runner / `ZmqStreamHelper` -- it could keep track of the cursor etc.?

Reply:
Changed